backend/middleware/authMiddleware.js

```diff
@@ -17,10 +17,9 @@ const authorize = (role) => {
         const decoded = jwt.verify(token, process.env.JWT_SECRET)
         req.user = await User.findById(decoded.id)
         const user = req.user
-        if (!user.roles.includes(role)) {
+        if (!role.includes(user.roles)) {
```

Comment:
Can you change the name of field according to their data type. Like since user.roles is a string, you should change it as user.role. Similarly for "role".